hudi-common/src/main/java/org/apache/hudi/metadata/BaseTableMetadata.java

```diff
@@ -344,8 +326,18 @@ protected HoodieEngineContext getEngineContext() {
     return engineContext != null ? engineContext : new HoodieLocalEngineContext(hadoopConf.get());
   }
 
+  public HoodieMetadataConfig getMetadataConfig() {
+    return metadataConfig;
+  }
+
   protected String getLatestDatasetInstantTime() {
     return datasetMetaClient.getActiveTimeline().filterCompletedInstants().lastInstant()
         .map(HoodieInstant::getTimestamp).orElse(SOLO_COMMIT_TIMESTAMP);
   }
+
+  @Override
+  public Option<String> getSyncedInstantTimeForReader() {
+    return Option.ofNullable(timelineMergedMetadata).map(TimelineMergedTableMetadata::getSyncedInstantTime());
```

Comment:
`TimelineMergedTableMetadata::getSyncedInstantTime` to get past compile issues. but still `getSyncedInstantTime()`, returns a option by itself. So, may be we need to indeed go back to the following?

```
@Override
  public Option<String> getSyncedInstantTimeForReader() {
    if (timelineMergedMetadata == null) {
      return Option.empty();
    }
    return timelineMergedMetadata.getSyncedInstantTime();
  }
```

Reply:
Oh.  Dont know why it compiles for me. I will fix it.